handlers.go

```diff
@@ -249,7 +250,9 @@ func (dht *IpfsDHT) handlePing(_ context.Context, p peer.ID, pmes *pb.Message) (
 }
 
 func (dht *IpfsDHT) handleFindPeer(ctx context.Context, p peer.ID, pmes *pb.Message) (*pb.Message, error) {
-	defer log.EventBegin(ctx, "handleFindPeer", p).Done()
+	ctx = log.Start(ctx, "handleFindPeer")
+	defer log.Finish(ctx)
+	log.LogKV(ctx, "peer", p.Pretty())
```

Comment:
This should probably be `log.SetTag(ctx, "peer", p)`.

1. `LogKV` is more for events/log items. `SetTag` will set the "peer" property for the event.
2. `Pretty` will convert to base58 (slow).

Reply:
I know lots of base58 encoding is too slow, but without this, the output will be:
```
{<...>, "Tags":{"peer":"\u0012 =\ufffd\ufffd\ufffd\n\ufffds}\ufffdGpnc0\ufffd\ufffd\ufffdhc\ufffd,}\ufffd\"\ufffd\u000c\ufffd\u0000ږ\ufffd\u0005","system":"dht"},"Logs":[{"Timestamp":"2018-08-24T10:51:23.56479413+08:00","Fields":[{"Key":"key","Value":"QmVYL1EPRThV5pg2Vybz2kxuCzyUcmvvRZXToQqs4aSxBn"}]}]}
```

Reply:
That looks like a bug. Our logger should be calling `String()`.

Reply:
Fixed here: https://github.com/ipfs/go-log/pull/45.